### ISSUE TITLE
Why is CT_2 so verbose

### ISSUE THREAD

**Issue by sloriot** [CGAL/cgal#2752]
In the code of `Constrained_triangulation_2` we have the following:

```
  std::cerr << "You are using an exact number types" << std::endl;
  std::cerr << "using a Constrained_triangulation_plus_2 class" << std::endl;
  std::cerr << "would avoid cascading intersection computation" << std::endl;
  std::cerr << " and be much more efficient" << std::endl;
```

in the body of the insert function. I don't see why we have this on.
I'm in a setting where I'm not supposed to have a lot of intersection of constraints so I'd like to stick to my CDT type and not move to CDT+

Do you agree that we should remove it?

**Comment by lrineau** [comment]
I understand why we want to educate our users to use better implementation. But could it be implemented by a compilation-time warning, instead of that run-time message?

**Comment by sloriot** [comment]
OK for a warning, maybe with a macro to disable it (like there use to be in Eigen to say that a package was experimental).

**Comment by lrineau** [comment]
Sébastien, you assigned the milestone 4.12 for this issue but, as the fix (#2785) involves a change in the documented API, I do not think we want that patch added between 4.12-beta1 and 4.12. Either we delay to 4.13-beta1, or we find an agreement quickly before 4.12-beta1.
